src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/SimpleWordTokenizer.cs

```diff
@@ -0,0 +1,208 @@
+/// </summary>
+internal static class SimpleWordTokenizer
+{
+    /// <summary>
+    /// Tokenizes the input text into individual words based on specific rules for text normalization and segmentation.
+    /// </summary>
+    /// <remarks>This method applies text normalization steps, such as removing skipped markers, handling line
+    /// breaks, and replacing common HTML entities. It also ensures consistent tokenization by inserting spaces around
+    /// punctuation, symbols, and certain character patterns. The tokenization rules are inspired by common BLEU algorithms,
+    /// such as those used in NLTK, SacreBLEU, and MOSES.</remarks>
+    /// <param name="text">The input text to be tokenized. Cannot be <see langword="null"/>.</param>
+    /// <returns>An enumerable collection of strings, where each string represents a tokenized word. The collection will be empty
+    /// if the input text contains no valid tokens.</returns>
+    public static IEnumerable<string> WordTokenize(string text)
+    {
+        _ = Throw.IfNull(text, nameof(text));
+
+        return WordTokenize(text.AsMemory());
+    }
+
+    /// <summary>
+    /// Tokenizes the input text into individual words based on specific rules for text normalization and segmentation.
+    /// </summary>
+    /// <remarks>This method applies text normalization steps, such as removing skipped markers, handling line
+    /// breaks, and replacing common HTML entities. It also ensures consistent tokenization by inserting spaces around
+    /// punctuation, symbols, and certain character patterns. The tokenization rules are inspired by common BLEU algorithms,
+    /// such as those used in NLTK, SacreBLEU, and MOSES.</remarks>
+    /// <param name="text">The input text to be tokenized. Cannot be <see langword="null"/>.</param>
+    /// <returns>An enumerable collection of strings, where each string represents a tokenized word. The collection will be empty
+    /// if the input text contains no valid tokens.</returns>
+    public static IEnumerable<string> WordTokenize(ReadOnlyMemory<char> text)
+    {
+        StringBuilder sb = new StringBuilder();
+
+        while (true)
+        {
+            if (text.IsEmpty)
+            {
+                if (sb.Length > 0)
+                {
+                    yield return sb.ToString();
+                    _ = sb.Clear();
+                }
+
+                yield break;
+            }
+
+            var span = text.Span;
+            char nextChar = span[0];
+
+            // Skip whitespace as separator
+            if (char.IsWhiteSpace(nextChar))
+            {
+                if (sb.Length > 0)
+                {
+                    yield return sb.ToString();
+                    _ = sb.Clear();
+                }
+
+                text = text.Slice(1);
+                continue;
+            }
+
+            // Join hyphenated words
+            if (span[0] == '-' &&
+                span.Length > 1 &&
+                span[1] == '\n')
```

Comment:
Do line-endings matter? Is it possible there is a `\r` before the `\n`? If yes, perhaps we need helpers to detect all kinds of line endings and skip the appropriate number of chars?

Reply:
Perhaps we can do something to keep skipping until we hit a character that is not `chat.IsWhiteSpace`...